packages/wrangler/src/index.tsx

```diff
@@ -108,7 +108,10 @@ async function printWranglerBanner() {
 }
 
 function isLegacyEnv(args: unknown, config: Config): boolean {
-  return config.legacy_env;
+  return (
+    (args as { "legacy-env": boolean | undefined })["legacy-env"] ??
```

Comment:
@JacobMGEvans Do you remember why this change was made in https://github.com/cloudflare/wrangler2/pull/680/files#diff-5e47c8c5ec00bfdc010f06653422c5db6be55b5c61226212cf593b8c57098b6cR103 ?

Reply:
This is because the `config.legacy_env` should have already taken into account if the `legacy-env` arg was being used.

See https://github.com/cloudflare/wrangler2/blob/b24aeb5722370c2e04bce97a84a1fa1e55725d79/packages/wrangler/src/config/validation.ts#L94-L98

@threepointone - was this actually causing an error somewhere? Could you provide a test that shows that the previous code was not working?

Reply:
Oh I see. Ok I'll add it as a comment then, I was confused why it was only reading `config`. We probably don't need to pass `args` to this function either then.